semaphore/semaphore.go

```diff
@@ -45,16 +46,16 @@ func (s *Weighted) Acquire(ctx context.Context, n int64) error {
 		return nil
 	}
 
+	var waiterList = &s.waiters
+
 	if n > s.size {
-		// Don't make other Acquire calls block on one that's doomed to fail.
-		s.mu.Unlock()
-		<-ctx.Done()
-		return ctx.Err()
+		// Add doomed Acquire call to the Impossible waiters list.
+		waiterList = &s.impossibleWaiters
```

Comment:
Wouldn't it make more sense to return sentinel error, ie ErrImpossibleWeight so that user can process it (especially since this implementation has Resize method)?

Reply:
This would be a breaking change to the behavior of the API